hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/HandleContext.java

```diff
@@ -110,6 +113,24 @@ enum TransactionCategory {
     @Nullable
     Key payerKey();
 
+    /**
+     * Get a calculator for calculating fees for the current transaction, and its {@link SubType}. Most transactions
+     * just use {@link SubType#DEFAULT}, but some (such as crypto transfer) need to be more specific.
+     *
+     * @param subType The {@link SubType} of the transaction.
+     * @return The {@link FeeCalculator} to use.
+     */
+    @NonNull
+    FeeCalculator feeCalculator(@NonNull final SubType subType);
+
+    /**
+     * Gets a {@link FeeAccumulator} used for collecting fees for the current transaction.
+     *
+     * @return The {@link FeeAccumulator} to use.
+     */
+    @NonNull
+    FeeAccumulator feeAccumulator();
```

Comment:
We probably will also have `FeeDistributor` here  to distribute fees to 800, 98 ? It is out of scope for this PR. Just curious.

Reply:
The `TokenServiceApiImpl` has the implementation of distribution based on API in `TokenServiceApi`